cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java

```diff
@@ -325,6 +306,8 @@ private static void validatePackageMdAndBalToml(Path balaPath) throws IOExceptio
                     return;
                 }
             }
+        } catch (IOException e) {
+            throw new ProjectException("error while validating the bala file.", e);
```

Comment:
This will again hide the original cause of the exception. Let's append the message of the orginal IOExceptioh here.

```suggestion
            throw new ProjectException("error while validating the bala file: " + e.getMessage(), e);
```

Reply:
Please check for other places that this happens.